python/pyarrow/array.pxi

```diff
@@ -538,6 +538,51 @@ def repeat(value, size, MemoryPool memory_pool=None):
     return pyarrow_wrap_array(c_array)
 
 
+def mask(indices, length, MemoryPool memory_pool=None):
```

Comment:
@jorisvandenbossche Does this API look ok?

Reply:
Late response, and the general API looks good, my one suggestion would be to use a bit more descriptive name. `mask` could also be interpret in the active sense ("mask those values"), and eg pandas has a method with that name that work in that sense. Some more explicit options: `create_mask`, `make_mask`, `mask_from_indices`, ..

This is essentially the counterpart for `indices_nonzero` I think? (which converts a mask into indices) 
Could maybe mention that in a "See Also" section in the docstring